src/os/ui/file/exportmanager.js

```diff
@@ -198,27 +196,56 @@ os.ui.file.ExportManager.prototype.doExport_ = function(items, fields, title, ex
 /**
  * Finishes the export when the exporter output is ready.
  *
- * @param {os.ex.IExportMethod} exporter The export method
- * @param {os.ex.IPersistenceMethod} persister The persistence method
+ * @param {os.ex.ExportOptions} options The export options.
  * @param {goog.events.Event=} opt_event The complete event
  * @private
  */
-os.ui.file.ExportManager.prototype.onExportComplete_ = function(exporter, persister, opt_event) {
+os.ui.file.ExportManager.prototype.onExportComplete_ = async function(options, opt_event) {
+  const exporter = options.exporter;
   var result = exporter.getOutput();
   var name = exporter.getName();
-  var extension = '.' + exporter.getExtension();
   exporter.dispose();
 
   if (result && name) {
-    // append the extension if it hasn't been already
-    if (!goog.string.endsWith(name, extension) && extension.length > 1) {
-      name += extension;
+    if (!options.keepTitle) {
+      // append the extension if it hasn't been already
+      var extension = '.' + exporter.getExtension();
+      if (!goog.string.endsWith(name, extension) && extension.length > 1) {
+        name += extension;
+      }
     }
 
-    persister.save(name, result, exporter.getMimeType());
+    if (options.persister) {
+      options.persister.save(name, result, exporter.getMimeType());
+    } else {
+      try {
+        if (result instanceof Blob) {
+          await result.arrayBuffer().then((arrayBuffer) => result = arrayBuffer);
```

Comment:
I wasn't aware of that, neat. 

I needed to revisit this function anyway because `async` functions piss of the compiler when they are passed as the handler to a `listen` function like this one is above. It's technically a type error because `async` functions always return a promise, which is doesn't match the allowed function type for the listen handler.